### PR TITLE
Support party specific grpc metadata

### PR DESCRIPTION
**Pull request by fengsp**

### CODE REVIEW

**Review by jovany-wang — APPROVED** [review]
LGTM @NKcqx and @zhouaihui CC as well.

File: tests/test_transport_proxy.py [hunk 1]
```diff
@@ -178,6 +178,41 @@ def test_send_grpc_with_meta():
...
+        }
+    }
+    compatible_utils.kv.put(constants.KEY_OF_CLUSTER_CONFIG,
+                            cloudpickle.dumps(cluster_config))
+    compatible_utils.kv.put(constants.KEY_OF_JOB_CONFIG,
+                            cloudpickle.dumps(job_config))
+
+    SERVER_ADDRESS = "127.0.0.1:12344"
+    party = 'test_party'
+    cluster_config = {'test_party': {'address': SERVER_ADDRESS, 'grpc_metadata': (('token', 'test-party-token'),)}}
```
Comment (NKcqx) [inline tests/test_transport_proxy.py:203]: `cluster_config` has already been defined, better merge them together

File: fed/api.py [hunk 2]
```diff
@@ -165,6 +171,7 @@ def init(
     compatible_utils.kv.initialize()
 
     cluster_config = {
+        # TODO: this store party specific conf, not only addresses, give it a better name?
```
Comment (NKcqx) [inline fed/api.py:174]: The discussion can be opened somewhere else but a comment ?

File: tests/test_transport_proxy.py (hunk 1 above)
Comment (fengsp) [reply tests/test_transport_proxy.py]: fixed with a more clear name

File: fed/api.py (hunk 2 above)
Comment (fengsp) [reply fed/api.py]: removed

File: tests/test_transport_proxy.py [hunk 3]
```diff
@@ -178,6 +178,46 @@ def test_send_grpc_with_meta():
...
+                            cloudpickle.dumps(cluster_config))
+    compatible_utils.kv.put(constants.KEY_OF_JOB_CONFIG,
+                            cloudpickle.dumps(job_config))
+
+    SERVER_ADDRESS = "127.0.0.1:12344"
+    party = 'test_party'
+    cluster_parties_config = {
+        'test_party': {
+            'address': SERVER_ADDRESS,
+            'grpc_metadata': (('token', 'test-party-token'),)
```
Comment (NKcqx) [inline tests/test_transport_proxy.py:206]: How does this test case make sure that this declaration actually takes effect?

**Review by NKcqx — CHANGES_REQUESTED** [review]

File: tests/test_transport_proxy.py (hunk 3 above)
Comment (fengsp) [reply tests/test_transport_proxy.py:216]: Fixed, now test recv proxy accpets expected_metadata and check it

**Review by NKcqx — APPROVED** [review]
LGTM

**Pull request merged**